Feat/multisource search

katm201 opened the pull request:
These updates include:

1. API middleware can now handle handle multi-source and multi-topic searches

2. removed a file that had an unused helper function

3. tests for the API middleware (everything passes)

Note:
I've formatted the multi-source search to minimize the issues caused by having source names and not codes, but it will still error for some that don't follow the standard pattern.

Ex. 'BBC News' => 'BBC-News' [works]
Ex. 'Google News Brasil' => 'Google-News-Brasil' [doesn't work, expects 'google-news-br']

pjminer3 commented:
Looks good!  My one comment is there might be room to improve our last test by using  `response.data.every( /* test typeof element === object */)` to test all all elements in the response array instead of just the first

katm201 commented:
Added a new jasmine module with better matchers library. Documentation here: https://www.npmjs.com/package/jasmine-expect

I also updated the tests Patrick flagged to make better use of the new matchers.
